app/src/main/java/com/example/plantsapp/presentation/ui/notification/NotificationWorker.kt

```diff
@@ -0,0 +1,119 @@
+package com.example.plantsapp.presentation.ui.notification
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.hilt.work.HiltWorker
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.bumptech.glide.Glide
+import com.example.plantsapp.R
+import com.example.plantsapp.domain.model.Plant
+import com.example.plantsapp.domain.model.Task
+import com.example.plantsapp.domain.repository.TasksRepository
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.runBlocking
+import java.util.Date
+
+@HiltWorker
+class NotificationWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted workerParams: WorkerParameters,
```

Comment:
Add your your class NotificationManager (or plantnotificationManager, or other...) that will handle notifications, instead of moving it into Worker
And move all notification-related to new NotificationManager